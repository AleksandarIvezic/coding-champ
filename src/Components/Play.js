import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import SelectCategory from './SelectCategory';
import mockCategories from '../Library/mockCategories';
import '../Styles/Play.css';

const Play = () => {
  const [selectedCategory, setSelectedCategory] = useState('');
  const handleChange = (e) => {
    setSelectedCategory(e.target.value);
  };

  return (
    <>
      <div className="select-category">
        <h3>Choose your tech</h3>
        <SelectCategory handleChange={handleChange} />
        {selectedCategory
      && (
      <h3>
        {mockCategories.find((category) => category.name === selectedCategory).description}
      </h3>
      )}
      </div>
      <div className="difficulty-level d-flex">
        <button type="button" className="category flex-centered"><span>easy</span></button>
        <button type="button" className="category flex-centered"><span>medium</span></button>
        <button type="button" className="category flex-centered"><span>hard</span></button>
      </div>
      <div>
        <button className="play-button" type="button">
          <Link to="game">
            Start the quiz
          </Link>
        </button>
      </div>
    </>
  );
};

export default Play;
