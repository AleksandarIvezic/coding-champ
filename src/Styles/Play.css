.play-button {
  background-color: #db3d44;
  font-size: 3rem;
  color: #fff;
  padding: 15px 15px;
  border-radius: 50%;
  border: none;
}

.play-button:hover {
  background-color: #d62d35;
  transform: scale(1.1);
  color: rgb(233, 206, 206);
  border-radius: 50%;
}
