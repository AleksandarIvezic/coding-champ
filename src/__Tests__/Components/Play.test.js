import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Play from '../../Components/Play';

it('renders correctly', () => {
  const tree = renderer
    .create(<Router><Play /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
