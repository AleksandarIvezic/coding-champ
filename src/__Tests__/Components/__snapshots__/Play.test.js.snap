// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="select-category"
  >
    <h3>
      Choose your tech
    </h3>
    <form>
      <label
        htmlFor="submit-category"
      >
        Choose a category
        <select
          defaultValue=""
          id="categories"
          name="categories"
          onChange={[Function]}
        >
          <option
            disabled={true}
            value=""
          >
             
          </option>
          <option
            value="HTML"
          >
            HTML
          </option>
          <option
            value="CSS"
          >
            CSS
          </option>
          <option
            value="JavaScript"
          >
            JavaScript
          </option>
        </select>
      </label>
    </form>
    
  </div>,
  <div
    className="difficulty-level d-flex"
  >
    <button
      className="category flex-centered"
      type="button"
    >
      <span>
        easy
      </span>
    </button>
    <button
      className="category flex-centered"
      type="button"
    >
      <span>
        medium
      </span>
    </button>
    <button
      className="category flex-centered"
      type="button"
    >
      <span>
        hard
      </span>
    </button>
  </div>,
  <div>
    <button
      className="play-button"
      type="button"
    >
      <a
        href="/game"
        onClick={[Function]}
      >
        Start the quiz
      </a>
    </button>
  </div>,
]
`;
